compiler-core/src/error.rs

```diff
@@ -394,6 +400,46 @@ numbers and underscores.",
                 write_project(buf, diagnostic);
             }
 
+            Error::InvalidLicense { license, reason } => {
+                let msg;
+                let diagnostic = ProjectErrorDiagnostic {
+                    title: "Invalid license".to_string(),
+                    label: format!(
+                        "The license \"{}\" {}.",
+                        license,
+                        match reason {
+                            spdx::error::Reason::UnknownLicense => "is not a SPDX license",
+                            spdx::error::Reason::UnknownException => "contains non-SPDX exception",
+                            spdx::error::Reason::InvalidCharacters => "contains invalid characters",
+                            spdx::error::Reason::UnclosedParens =>
+                                "contains an opening parenthesis without a closing parenthesis",
+                            spdx::error::Reason::UnopenedParens =>
+                                "contains a closing parenthesis without an opening parenthesis",
+                            spdx::error::Reason::Empty =>
+                                "contains no terms that are valid in SPDX expressions",
+                            spdx::error::Reason::Unexpected(expected) => match expected.len() {
+                                0 => "contains an unexpected term",
+                                1 => {
+                                    msg = String::from("contains an unexpected term, expected ") + expected.first().unwrap();
```

Comment:
I think it is ideal to avoid unwrap if possible? Maybe `expected.first().unwrap_or("[unknown]")` or some similar default value that we can probably rely on never return. Or I guess (though I've not done much of it rust) is that you can match on `expected` as a slice rather than matching on its length and then you can match against an empty slice and a single element slice and then `_` so that in the single element slice situation you have direct access to the element itself. A bit like the first example on this page: https://adventures.michaelfbryan.com/posts/daily/slice-patterns/

Reply:
Ah, right, I originally did the `unwrap` because I thought `expected` was a `Vec` and (IIRC) you can't pattern-match on `Vec`s in a `match`. However I was wrong; as you note, it is a slice.